src/api/helpers.jl

```diff
@@ -361,18 +361,26 @@ function h5p_get_efile_prefix(plist)
 end
 
 function h5p_get_external(plist, idx = 0)
-    name_size = Csize_t(1024)
-    name = zeros(UInt8, name_size)
-    offset = Ref{off_t}(0)
+    offset = Ref{Int64}(0)
     sz = Ref{hsize_t}(0)
-    h5p_get_external(plist, idx, name_size, name, offset, sz)
-    # name may not be null terminated according to H5P_GET_EXTERNAL documentation
-    nul_idx =  findfirst(==(0x00), name)
-    name_size = nul_idx === nothing ? name_size : nul_idx-1
+    name_size = 64
+    name = Base.StringVector(name_size)
+    null_id = nothing
+    while true
+        h5p_get_external(plist, idx, name_size+1, name, offset, sz)
```

Comment:
There might be an alternative way to get the name through a different hdf5 method. 

For example, that's what we did here:
https://github.com/JuliaIO/HDF5.jl/blob/master/src/api/helpers.jl#L591

Reply:
Why `name_size+1`? The library is going to overrun the buffer.

Reply:
```suggestion
        h5p_get_external(plist, idx, name_size, name, offset, sz)
```

Reply:
```suggestion
        h5p_get_external(plist, idx, name_size, name, offset, sz)
```

Reply:
Stringvector includes an implied null

Reply:
It might be - not +, but I'll check later.

Reply:
The size passed in the library needs to include a slot for the null byte. Whereas when you use string vector the length does not include the null byte (it's implied). This is why we need to pass in `name_size+1`.

Reply:
Let me double check that for this method, the above even matters since it's not returning the required buffer size for this hdf5 method.

Reply:
Ok right this case it doesn't actually matter since the library wants the full array length.